src/main/java/org/assertj/core/groups/Tuple.java

```diff
@@ -12,45 +12,38 @@
  */
 package org.assertj.core.groups;
 
-import static java.util.Collections.addAll;
 import static org.assertj.core.configuration.ConfigurationProvider.CONFIGURATION_PROVIDER;
-import static org.assertj.core.util.Lists.newArrayList;
-import static org.assertj.core.util.Objects.areEqual;
 
+import java.util.Arrays;
 import java.util.List;
 
 public class Tuple {
 
-  private final List<Object> datas = newArrayList();
+  private final List<Object> values;
 
   public Tuple(Object... values) {
-    addAll(datas, values);
+    this.values = Arrays.asList(values);
   }
 
   public Object[] toArray() {
-    return datas.toArray();
+    return values.toArray();
   }
 
   public List<Object> toList() {
-    return datas;
+    return values;
   }
 
   @Override
-  public int hashCode() {
-    final int prime = 31;
-    int result = 1;
-    result = prime * result + datas.hashCode();
-    return result;
-  }
-
-  @Override
-  public boolean equals(Object obj) {
+  public final boolean equals(Object obj) {
```

Comment:
The `final` on `equals` and `hashCode` is [a suggestion from EqualsVerifier](https://jqno.nl/equalsverifier/errormessages/subclass-equals-is-not-final/), but there are other strategies if we don't want to be so strict.

Reply:
I'm ok with that

Reply:
I'm ok with that, we might even put the class as final, I don't see why people would subclass `Tuple` it's really an immutable data class.